src/dataSources/JSON.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+var Null = require('./Null');
+
+/**
+ * @constructor
+ */
+var JSON = Null.extend('JSON',  {
+
+    initialize: function(data, fields, calculators) {
+        this.setData(data, fields, calculators);
```

Comment:
Yes currently the init of Behavior and DM calls setData. I need to fix those first